src/libcore/slice/mod.rs

```diff
@@ -841,6 +841,41 @@ impl<T> [T] {
         ChunksExactMut { v: fst, rem: snd, chunk_size }
     }
 
+    /// Returns an iterator over `N` elements of the slice at a time, starting at the
+    /// beginning of the slice.
+    ///
+    /// The chunks are slices and do not overlap. If `N` does not divide the length of the
+    /// slice, then the last up to `N-1` elements will be omitted and can be retrieved
+    /// from the `remainder` function of the iterator.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `N` is 0.
```

Comment:
Could you please add a note here that says that future versions of this are allowed to panic at compile-time if `N` is 0 instead of run-time?

Reply:
While that's desirable, I am not sure how feasible it is to change this after stabilization.

Opened a discussion on zulip about this: https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/array_chunks.20of.20size.200/near/204001667

Reply:
Ended up changing `array_chunks::<0>()` to a compile time error. While we may removing this bound before this method hits stable, it seems like the better default for now.
```
error[E0277]: the trait bound `[(); 0]: core::slice::sealed::NonZero` is not satisfied
```

Reply:
FWIW, there's also an upper bound on array lengths for any non-ZST `T`, but that should get a compile-time ``error: the type `...` is too big for the current architecture``.

Reply:
Aaah, nm. Adding this bound breaks type inference. Looks like we may have to live with a runtime error for now